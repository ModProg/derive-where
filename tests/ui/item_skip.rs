use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where(Debug)]
#[derive_where(skip_inner)]
enum SkipInnerOnEnum<T> {
	A(PhantomData<T>),
}

#[derive_where(Debug)]
#[derive_where(skip_inner)]
#[derive_where(skip_inner)]
struct DuplicateSkipInner<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner())]
struct EmptySkipInner<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner)]
#[derive_where(skip_inner(Debug))]
struct OverridingSkipInner<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner(Debug))]
#[derive_where(skip_inner)]
struct UnderridingSkipInner<T>(PhantomData<T>);

#[derive_where(Clone; T)]
#[derive_where(skip_inner)]
struct NoSupportedTrait<T>(PhantomData<T>);

#[derive_where(Copy; T)]
#[derive_where(skip_inner(Copy))]
struct UnsupportedTrait<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner(Debug, Debug))]
struct DuplicateTraitSame<T>(PhantomData<T>);

#[derive_where(Debug)]
#[derive_where(skip_inner(Debug))]
#[derive_where(skip_inner(Debug))]
struct DuplicateTraitSeparate<T>(PhantomData<T>);

#[derive_where(Clone; T)]
#[derive_where(skip_inner(Debug))]
struct MissingDeriveTrait<T>(PhantomData<T>);

fn main() {}
