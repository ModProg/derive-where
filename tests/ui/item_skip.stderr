error: enums don't support `skip_inner`, use it on a variant instead
 --> tests/ui/item_skip.rs:6:16
  |
6 | #[derive_where(skip_inner)]
  |                ^^^^^^^^^^

error: duplicate `skip_inner` option
  --> tests/ui/item_skip.rs:13:16
   |
13 | #[derive_where(skip_inner)]
   |                ^^^^^^^^^^

error: empty attribute option found
  --> tests/ui/item_skip.rs:17:16
   |
17 | #[derive_where(skip_inner())]
   |                ^^^^^^^^^^^^

error: unexpected constraint on `skip` when unconstrained `skip` already used
  --> tests/ui/item_skip.rs:22:16
   |
22 | #[derive_where(skip_inner(Debug))]
   |                ^^^^^^^^^^^^^^^^^

error: duplicate `skip_inner` option
  --> tests/ui/item_skip.rs:27:16
   |
27 | #[derive_where(skip_inner)]
   |                ^^^^^^^^^^

error: no trait that can be skipped is being implemented
  --> tests/ui/item_skip.rs:31:16
   |
31 | #[derive_where(skip_inner)]
   |                ^^^^^^^^^^

error: unsupported skip group, expected one of Clone, Debug, EqHashOrd, Hash, Zeroize
  --> tests/ui/item_skip.rs:35:27
   |
35 | #[derive_where(skip_inner(Copy))]
   |                           ^^^^

error: duplicate `Debug` constraint on `skip`
  --> tests/ui/item_skip.rs:39:34
   |
39 | #[derive_where(skip_inner(Debug, Debug))]
   |                                  ^^^^^

error: duplicate `Debug` constraint on `skip`
  --> tests/ui/item_skip.rs:44:27
   |
44 | #[derive_where(skip_inner(Debug))]
   |                           ^^^^^

error: trait to be skipped isn't being implemented
  --> tests/ui/item_skip.rs:48:27
   |
48 | #[derive_where(skip_inner(Debug))]
   |                           ^^^^^
